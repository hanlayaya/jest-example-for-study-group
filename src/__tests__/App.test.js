import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('<App />', function () {
  let App;

  beforeEach(() => {
    jest.resetAllMocks();
    App = require('../App').default;

    this.makeSubject = () => {
      return shallow(
        <App />
      );
    };
  });

  describe('render App', () => {
    beforeEach(() => {
      this.subject = this.makeSubject();
    });

    it('should be same as snapshot', () => {
      expect(toJson(this.subject)).toMatchSnapshot();
    });
  })
});
