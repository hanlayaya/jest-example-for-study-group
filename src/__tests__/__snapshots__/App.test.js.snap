// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> render App should be same as snapshot 1`] = `
<div
  className="App"
>
  <div
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h2>
      前端找工作
    </h2>
    <Connect(Timer) />
  </div>
  <Connect(JobList) />
</div>
`;
